src/suffix_array.cc

```diff
@@ -70,12 +64,13 @@ namespace fuzzy
     std::vector<std::vector<SuffixView> > prefixes_by_word_id(vocab_size);
 
     // sort suffixes according to their first word id
-    for (size_t i = 0; i < _suffixes.size(); i++)
+    for (const auto& suffix : _suffixes)
     {
-      const auto wid = (*this)[_suffixes[i]];
+      const auto* suffix_wids = get_suffix(suffix);
```

Comment:
I find that it would be better on one line.

`const auto wid = get_suffix(suffix)[0];`